### DISCUSSION TITLE
What features would you like to see?

### DISCUSSION THREAD

**Discussion by yupix** [yupix/Mi.py#66] [Ideas]
Please write down the features you want here. We'll move it to an issue when we have a full-fledged idea.

**Comment by fotoente** [comment]
Not sure if its fully implemented yet:
A automatic reply to a mention no matter which user or instance it comes from

**Reply by yupix** [reply]
The on_mention event may have been present in v0.2.5, but it is not working now due to internal changes to v1.0.0, and will be implemented in v1.0.1.

**Reply by yupix** [reply]
on_mention event is added v1.0.1
https://github.com/yupix/Mi.py/releases/tag/v1.0.1

**Comment by fotoente** [comment]
Is there a way to see how many replies there have been on some post?
Its just to prevent the bot to reply until the end of time when it interacts with some other bot (e.g. it stops after may ten replies)

**Reply by yupix** [reply]
There is currently no function to get the total number of replies to a note. I'll let you know in Misskey's DM as soon as I confirm that I can create that feature tomorrow or so.

**Reply by yupix** [reply]
Mi.py v2.0.0 adds get_replies and get_note methods to Note class.
Since the get_replies method can get the list of replies in List[Note] format, we can get the number of replies with the following code.
```python
replies = await self.get_replies('note_id')
print(len(replies))
```

**Reply by fotoente** [reply]
Thank you for your response.
I think i worded wrong what i was looking for.

I was looking for a function to get the length of the thread. That is posted.

For example:
Bot A & Bot B interacts with each other. If there isn't a count on how many answers there have been in one thread it would go on like forever. So it would be nice if there could be away to e.g. check if some thread length has been exceeded and then the bot stops answering on mentions in this thread.

As i said, it is not a problem with users, but with other bots and to just stop interacting with usera ccounts with the Bot-Flag set (as i do in the moment) excludes some users that have this flag set (for reasons unknown to me)

**Reply by yupix** [reply]
By thread length, do you mean replies to replies? Am I right in assuming that it looks like the following image? If not, I would appreciate it if you could send me the actual image.
![image](https://user-images.githubusercontent.com/50538210/147670939-96972e47-317b-4aa8-8b88-44c369dc2e65.png)

**Reply by yupix** [reply]
I'm going to bed now, so please understand in advance that my reply will be late.

**Reply by fotoente** [reply]
Yes it looks like that.
A bot answering a bot and without a limit of some sort (like the number of notes in a thread) it would go on like forever
![botstalking](https://user-images.githubusercontent.com/96478337/147741703-57f93fe8-5bdf-49fc-9a7e-9267a583ba0a.png)

Here is the link so you can see what I am talking about: https://ente.fun/notes/8usfd68ecl

In the end the bot named `feli_bot` has a limit implemented, so it ended after about 10 notes, but if there wasn't such a limit the bots would still talk today.

**Reply by yupix** [reply]
Hmmm... It looks like I need to figure out a way to get a series of replies to a note, so I can't promise to implement it right now, but I'd like to add it as a new feature by v3.0.0.
